src/common/stats/StatsManager.h

```diff
@@ -17,6 +17,40 @@
 namespace nebula {
 namespace stats {
 
+class MetricsSerializer {
+public:
+    virtual ~MetricsSerializer() = default;
+    std::string serialize() /*const*/ {
+        std::ostringstream ss;
+        serialize(ss);
+        return ss.str();
+    }
+    virtual void serialize(std::ostream& out) /*const*/ = 0;
+};
+
+class PrometheusSerializer : public MetricsSerializer {
+public:
+    virtual ~PrometheusSerializer() = default;
+    void serialize(std::ostream& out) /*const*/ override {
+        prometheusSerialize(out);
+    }
+    virtual void prometheusSerialize(std::ostream& out) /*const*/ = 0;
+    struct Label {
+        std::string name;
+        std::string value;
+    };
+    void annotate(std::ostream& out, std::string& metric_name, std::string&& metric_type) const;
+    void writeValue(std::ostream& out, double value) const;
+    void writeValue(std::ostream& out, const int64_t value) const;
+    void writeValue(std::ostream& out, const std::string& value) const;
+    template <typename T = std::string>
+    void writeHead(std::ostream& out, const std::string& family,
+                const std::vector<Label>& labels, const std::string& suffix = "",
+                const std::string& extraLabelName = "",
+                const T extraLabelValue = T()) const;
```

Comment:
T is value type, so this `const` of parameter type T is not necessary.

Reply:
The parameter is const too.